src/main/java/com/epam/aidial/core/service/ResourceTopic.java

```diff
@@ -0,0 +1,87 @@
+package com.epam.aidial.core.service;
+
+import com.epam.aidial.core.data.ResourceEvent;
+import com.epam.aidial.core.storage.ResourceDescription;
+import io.vertx.core.impl.ConcurrentHashSet;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.redisson.api.RTopic;
+import org.redisson.api.RedissonClient;
+import org.redisson.codec.TypedJsonJacksonCodec;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Consumer;
+
+
+@Slf4j
+public class ResourceTopic {
+
+    private final Map<String, Set<Subscription>> urlToSubscriptions = new ConcurrentHashMap<>();
```

Comment:
keys are never evicted

Reply:
Check Map contract and ConcurrentHashMap implementation.